client/pkg/infra/controllers/provision.go

```diff
@@ -106,15 +106,34 @@ func (ctrl *ProvisionController[T]) Settings() controller.QSettings {
 }
 
 // MapInput implements controller.QController interface.
-func (ctrl *ProvisionController[T]) MapInput(_ context.Context, _ *zap.Logger,
-	_ controller.QRuntime, ptr resource.Pointer,
+func (ctrl *ProvisionController[T]) MapInput(ctx context.Context, _ *zap.Logger,
+	r controller.QRuntime, ptr resource.Pointer,
 ) ([]resource.Pointer, error) {
 	if ptr.Type() == siderolink.ConnectionParamsType {
 		return nil, nil
 	}
 
+	id := ptr.ID()
```

Comment:
should it be more explicit about types of the inputs here?

as it's not clear what happens in the flow if the type is ConfigPatchRequestType?

might be `switch ptr.Type() {` ?

Reply:
ok, let's change it to switch :)